libraries/cyclestreets-view/src/main/java/net/cyclestreets/routing/Segments.kt

```diff
@@ -44,16 +44,56 @@ class Segments : Iterable<Segment> {
             if (previous.distance < 20) {
                 CROSSROAD_MELDS[Pair(previous.turn, seg.turn)]?.let {
                     segments.remove(previous)
-                    segments.add(Segment.Step(previous, seg, it, it.textInstruction))
-                    return
+                    if (previous.walk == seg.walk) {
+                        // Combine the 2 segments:
+                        segments.add(Segment.Step(previous, seg, it, it.textInstruction))
+                        return
+                    }
+                    else {
+                    // If the 2 segments aren't both walking or both cycling, keep the segments separate.
+                    // Add previous segment with combined turn instruction and CURRENT street,
+                    // so user still gets warning of the double turn:
+                        segments.add(Segment.Step(
```

Comment:
Massively redundant block here.  Just move `segments.remove()` into the `if` branch and then you don't need an `else` branch, and all its noisy comments, at all.

Reply:
In the else branch, I'm effectively modifying the previous segment to change the previous instruction and street, so the user still gets warning of the double turn.
If I put the remove inside the if branch and take out the else branch, the user won't get warning of the double turn.  I thought it would be nice to still have the double turn instruction, even for mixed cycling / walking (esp where the first segment of the 2 is a cycling one).